libs/common/src/lib/inject-map-api.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import { SettingsService } from '@placeos/common';
+import { BehaviorSubject, Observable, Subject } from 'rxjs';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class InjectMapApiService {
+    private maps_key_exists = new BehaviorSubject<boolean>(null);
+
+    private _setKeyExists(value: boolean) {
+        this.maps_key_exists.next(value);
+    }
+
+    public getKeyExistsAsObservable(): Observable<boolean> {
+        return this.maps_key_exists.asObservable();
+    }
+
+    constructor(private _settings: SettingsService) {}
+
+    public injectMapsApiKeys() {
+        const maps_key = this._settings.get('app.maps_indoor_api_key');
+        if (maps_key) {
+            const script = document.createElement('script');
+            script.src = `https://app.mapsindoors.com/mapsindoors/js/sdk/4.21.4/mapsindoors-4.21.4.js.gz?apikey=${maps_key}`;
+            document.head.appendChild(script);
```

Comment:
Probably doesn't matter after the page has loaded but add scripts to the bottom of the body element.

Reply:
I've updated this now.